### PR TITLE
Polio-1206:afro map list view

### PR DESCRIPTION
**Pull request by quang-le**
Nobody asked for this, I got another ticket wrong. Bonus for the client

Related JIRA tickets :  POLIO-1187, POLIO-1206

## Self proofreading checklist

- [X] Did I use eslint and black formatters
- [X] Is my code clear enough and well documented
- [ ] Are my typescript files well typed
- [ ] New translations have been added or updated if new strings have been introduced in the frontend
- [ ] My migrations file are included
- [ ] Are there enough tests
- [ ] Documentation has been included (for new feature)

## Changes

- backend: add some fields to payload, remove exclsuion of shapeless districts
- frontend:
    - Add tabs to the afro map
    - refactor components to be able to keep round selector in bot map and list view
    - reused API hooks for list view with ame key to avoid double fetching
    - add a context to pass the map bounds to the list view (for the API calls)
    - use css hack to prevent page from crashing when loading with 2 list views (deep linked navigation)

## How to test

Go to LQAS afro map, play with the tabs

## Print screen / video

https://github.com/BLSQ/iaso/assets/38907762/59488d48-033d-4645-9756-0631103029a3

##Notes

Needs POLIO-1177

### CODE REVIEW

File: plugins/polio/js/src/domains/LQAS-IM/LQAS/LqasAfroOverview/ListView/HasLocationIcon.tsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    withLocation: {
+        defaultMessage: 'With point',
+        id: 'iaso.registry.withLocation',
+    },
+    noGeographicalData: {
+        defaultMessage: 'Without geography',
+        id: 'iaso.label.noGeographicalData',
+    },
+});
+type Props = { shape: any };
```
Comment (beygorghor) [inline plugins/polio/js/src/domains/LQAS-IM/LQAS/LqasAfroOverview/ListView/HasLocationIcon.tsx]: Shape type is here: hat/assets/js/apps/Iaso/domains/orgUnits/types/shapes.ts

File: plugins/polio/js/src/domains/LQAS-IM/LQAS/LqasAfroOverview/ListView/LqasAfroList.tsx [hunk 2]
```diff
@@ -0,0 +1,372 @@
...
+            result = sortBy(result, ['data.district_name']);
+        } else if (sortFocus === 'STATUS') {
+            result = sortBy(result, ['status']);
+        } else if (sortFocus === 'ROUND') {
+            result = sortBy(result, ['data.round_number']);
+        } else if (sortFocus === 'CAMPAIGN') {
+            result = sortBy(result, ['data.campaign']);
+        } else if (sortFocus === 'LOCATION') {
+            result = sortBy(result, ['data.geo_json']);
+        }
```
Comment (beygorghor) [inline plugins/polio/js/src/domains/LQAS-IM/LQAS/LqasAfroOverview/ListView/LqasAfroList.tsx]: A switch is maybe cleaner for this no ?

File: plugins/polio/js/src/domains/LQAS-IM/LQAS/LqasAfroOverview/Map/LqasAfroMap.tsx [hunk 3]
```diff
@@ -12,14 +12,15 @@ import { AfroMapParams, Side } from '../types';
...
+import { LqasAfroOverviewContext } from '../Context/LqasAfroOverviewContext';
 
 type Props = {
     router: Router;
     side: Side;
 };
 
 export const LqasAfroMap: FunctionComponent<Props> = ({ router, side }) => {
-    const [bounds, setBounds] = useState<Bounds | undefined>(undefined);
+    const { bounds, setBounds } = useContext(LqasAfroOverviewContext);
```
Comment (beygorghor) [inline plugins/polio/js/src/domains/LQAS-IM/LQAS/LqasAfroOverview/Map/LqasAfroMap.tsx]: ❤️

File: plugins/polio/js/src/domains/LQAS-IM/LQAS/LqasAfroOverview/Map/LqasAfroMapPanesContainer.tsx [hunk 4]
```diff
@@ -1,8 +1,9 @@
 import React, {
     FunctionComponent,
-    useState,
+    // useState,
```
Comment (beygorghor) [inline plugins/polio/js/src/domains/LQAS-IM/LQAS/LqasAfroOverview/Map/LqasAfroMapPanesContainer.tsx]: 🗑️

**Review by beygorghor — CHANGES_REQUESTED** [review]
Please provide a message saying you have nothing to show
Screenshot 2023-10-04 at 11 40 58
I would remove the separator now if you have tabs
Screenshot 2023-10-04 at 11 41 12
I have missing translations:
Screenshot 2023-10-04 at 11 42 41

File: plugins/polio/js/src/domains/LQAS-IM/LQAS/LqasAfroOverview/ListView/LqasAfroList.tsx (hunk 2 above)
Comment (quang-le) [reply plugins/polio/js/src/domains/LQAS-IM/LQAS/LqasAfroOverview/ListView/LqasAfroList.tsx]: I don't know, I find `switch` to be needlessly verbose

File: plugins/polio/js/src/domains/LQAS-IM/LQAS/LqasAfroOverview/ListView/HasLocationIcon.tsx (hunk 1 above)
Comment (quang-le) [reply plugins/polio/js/src/domains/LQAS-IM/LQAS/LqasAfroOverview/ListView/HasLocationIcon.tsx]: Not the right type, I'm afraid, but there was already a type in polio constants 😀

**Review by beygorghor — APPROVED** [review]

**Pull request merged**